### PR TITLE
chore(gamut-system): renamed some of the createHandler internals

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- ~[ ] Related to designs:~
- ~[ ] Related to JIRA ticket: ABC-123~
- [ ] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

* Rename `altProps` to `dependentProps`
* Rename `templateFns` to `styleTemplates`
* Renames the functions to generate ☝️  from a `*Property* format to a `create*StyleTemplate` format
* Uses `Object.assign` instead of casting when returning functions with members on them

### CODE REVIEW

File: packages/gamut-system/src/core/compose/index.ts [hunk 1]
```diff
@@ -16,30 +11,26 @@ export const compose = <
...
-  const composedHandler = responsiveProperty<AbstractTheme, Props>(
-    config as any
-  ) as Handler<Props>;
-
   // Make the handlers propNames and functions accessible on the function reference
-  composedHandler.propNames = config.propNames;
-  composedHandler.templateFns = config.templateFns;
-
-  return composedHandler;
+  return Object.assign(createResponsiveStyleTemplate<Props>(config), config);
```
Comment (codecaaron) [inline packages/gamut-system/src/core/compose/index.ts:35]: This is much cleaner :)

File: packages/gamut-system/src/core/createHandler/index.ts [hunk 2]
```diff
@@ -25,41 +24,39 @@ export const createHandler = <
...
+  // These are all the possible prop names that this function could be responsible for.
+  // These will be passed to style functions to indicate which props should be computed.
+  const propNames = [propName, ...dependentProps] as (keyof Props)[];
 
-  // Create the style template function for the relative props
-  const styleFunction = templateFunction<
-    Props,
-    Config & { computeValue: TransformValue }
-  >({
+  // Use the requested template generator to create the style template fnction.
```
Comment (codecaaron) [inline packages/gamut-system/src/core/createHandler/index.ts:40]: typo :) 

```suggestion
  // Use the requested template generator to create the style template function.
```

**Review by codecaaron — APPROVED** [review]
This is great, much more readable and grokkable, thank you <3

One typo but not blocking :)

File: packages/gamut-system/src/core/createHandler/index.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-system/src/core/createHandler/index.ts:40]: eeeek

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.2.2-alpha.a9fe1a.0<br/>@codecademy/gamut-styles@7.1.0-alpha.a9fe1a.0<br/>@codecademy/gamut-system@0.0.1-alpha.a9fe1a.0<br/>@codecademy/gamut@16.0.12-alpha.a9fe1a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f7e1a4457fcf6009cde8209--gamut-preview.netlify.app>https://5f7e1a4457fcf6009cde8209--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f7e1a4457fcf6009cde8209>Deploy Logs</a></p>

**Comment by JoshuaKGoldberg** [comment]
@codecaaron nice fix with the last commit -- am I good to go to merge this? 🚀

**Pull request merged**